### ISSUE TITLE
OpenVMS install fails in execution phase

### ISSUE THREAD

**Issue by lgh-127001** [lenticularis39/axpbox#50]
Hello,
I followed the guide here in the repo about installing OpenVMS (my version is VSI OpenVMS 8.4l1), and the installation proceeds as described until the execution phase. The two documented error messages (about the hlb file and the exe file) are displayed, I dismiss them with a "no" for the question about terminating the install. After a few minutes however, at that time the progress was on 10%, the following messages appeared:
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYS$I18N.LOCALES.SYSTEM]UTF8-20.LOCALE;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYS$I18N.LOCALES.SYSTEM]UTF8-20.LOCALE;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYS$I18N.LOCALES.SYSTEM]UTF8-20.LOCALE;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYS$I18N.LOCALES.SYSTEM]UTF8-20.LOCALE;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-CLOSEOUT, error closing DISK$ALPHASYS:[VMS$COMMON.][SYS$I18N.LOCALES.SYSTEM]UTF8-20.LOCALE;1 as output 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-OPFAILED, operation failed 
Terminating is strongly recommended. Do you want to terminate? [YES] 
I don't know too much about the system yet, I'm just starting to learn about VMS, so I don't know what could cause the problem. Could you help me please, what could go wrong which causes these messages? After I dismiss them with "no" it starts to throw messages about nonexistent directories and I killed the process after the 10th, so don't know what would be the result at the end. I get at booting messages like "ide0.1 aborting on command 0x00" and the same message with "... command 0xec" is it possible this causes the problem? The config file is the same as in the guide, except that I replaced my file paths for disk files.
Thank you!

**Comment by RaymiiOrg** [comment]
I've edited your question, only added formatting to the error message.

What host os are you using and which version of axpbox ( git info is in the help output)?

Does the error also occur when you install VSI 8.4-2L2? (Or if you have access to it, the latest hp version?)

**Comment by lgh-127001** [comment]
Hello,
I'm using the following version on a Debian 10.6 WSL (but everything else runs fine on WSL so I don't think that causes this, however I can try in a native Debian too):
AXPBox Alpha Emulator (commit eb9dad4fb9eaf638e7bb2c3f23d3fbefa1403d51)
Unfortunately I have access only to VSI's community license so I have only the 8.4L1 version, I couldn't find anything about L2 on their FTP.
Or L2 is provided in the community license too?
Thank you!

**Comment by RaymiiOrg** [comment]
Oh that's interesting, WSL! Is it WSL 1 or 2? I did read [this](https://news.ycombinator.com/item?id=25612962) recently regarding file system corruption and WSL. 

If you could try it on a non wsl debian and see if the problem still occurs that would be helpful in reproducing the issue for me. 



On my guide I've linked to the hp openvms iso and an expired hobbyist license: https://raymii.org/s/tutorials/Installing_OpenVMS_8.4_Alpha_in_AXPbox_with_networking.html#toc_2 which you could try. But for the installation you don't need a license.

**Comment by lgh-127001** [comment]
Hello,
It was a WSL2, but now I tried on native Debian too, and it's very interesting because I got the same problem, just with other files, and the installation progressed a bit further before it crashed. I haven't tried the HPE version yet, but I will try it too and I'll send a comment about the results. The native Debian install results, with the exactly same version of the axpbox binary was the following, again in execution phase:
Portion done: 10%...20%...30%...40% 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYSLIB]BASIC$STARLET.TLB;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYSLIB]BASIC$STARLET.TLB;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYSLIB]BASIC$STARLET.TLB;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-WRITEERR, error writing DISK$ALPHASYS:[VMS$COMMON.][SYSLIB]BASIC$STARLET.TLB;1 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-CLOSEOUT, error closing DISK$ALPHASYS:[VMS$COMMON.][SYSLIB]BASIC$STARLET.TLB;1 as output 
-SYSTEM-F-CTRLERR, fatal controller error 
-SYSTEM-W-RESULTOVF, resultant string overflow 
%PCSI-E-OPFAILED, operation failed 
Terminating is strongly recommended. Do you want to terminate? [YES] 
I'm getting the same aborting on command errors on native Debian on startup too, I'm not sure if it has anything to do with this issue.
Thank you!

**Comment by lgh-127001** [comment]
Hello,
I had time now to try to install the HP version of OpenVMS you linked here. The interesting thing is, that it failed again, but with a completely different error, although again in the execution phase. I copied the errors below:
Portion done: 0%...10%...20%...30%...40%...50% 
%PCSI-E-READERR, error reading DQA1:[VMS$COMMON.][SYSLIB]TPU$MOTIFSHR.EXE;1 
-SYSTEM-F-DEVACTIVE, device is active 
-SYSTEM-W-PARITY, parity error 
%PCSI-E-READERR, error reading DQA1:[VMS$COMMON.][SYSLIB]TPU$MOTIFSHR.EXE;1 
-SYSTEM-F-DEVACTIVE, device is active 
-SYSTEM-W-PARITY, parity error 
%PCSI-E-READERR, error reading DQA1:[VMS$COMMON.][SYSLIB]TPU$MOTIFSHR.EXE;1 
-SYSTEM-F-DEVACTIVE, device is active 
-SYSTEM-W-PARITY, parity error 
%PCSI-E-READERR, error reading DQA1:[VMS$COMMON.][SYSLIB]TPU$MOTIFSHR.EXE;1 
-SYSTEM-F-DEVACTIVE, device is active 
-SYSTEM-W-PARITY, parity error 

%PCSI-E-OPENIN, error opening DQA1:[VMS$COMMON.][SYSLIB]TPU.DAT; as input 
-RMS-E-ACC, ACP file access failed 
-SYSTEM-F-DEVACTIVE, device is active 
%PCSI-E-OPFAILED, operation failed 
Terminating is strongly recommended. Do you want to terminate? [YES] 
After hitting some "no"'s it started to give the same messages about device is active, so at the end I killed the install process. I used the iso from the first link of your installation guide. This installation was on the same native debian like the previous install, so it wasn't the WSL.

**Comment by RaymiiOrg** [comment]
I've edited your comments to give different formatting to the error messages. I'll try and see if I can reproduce the issue with the current build. Thank you so far for the reports!

**Comment by fsword7** [comment]
That looks like IDE emulation.  Try SCSI emulation.

**Comment by RaymiiOrg** [comment]
This could indeed be related to https://github.com/lenticularis39/axpbox/issues/8 - as @fsword7 says you could try to change the disk device type to SCSI. Either the CD-ROM to install or the disk file you install on, or both.

**Comment by lgh-127001** [comment]
Hello,
I had time only now to try to install VSI OpenVMS with SCSI, and it solved all problems mentioned above. I could successfully install and boot the system. I can use it and install things on it!
thank you very much!

**Comment by kerk1v** [comment]
Yes, it seems IDE is somewhat broken. I am now trying to install with SCSI disks. 
SCSI disk speed also seems better than IDE, would be nice and make for a quicker OpenVMS install (which takes like 90 minutes anyways)
